heapster-saw/src/Verifier/SAW/Heapster/SAWTranslation.hs

```diff
@@ -3079,21 +3102,28 @@ translatePermImpl1 :: ImplTranslateF r ext blocks tops ret =>
 translatePermImpl1 prx mb_impl mb_impls = case (mbMatch mb_impl, mbMatch mb_impls) of
   -- A failure translates to a call to the catch handler, which is the most recent
   -- Impl1_Catch, if one exists, or the SAW errorM function otherwise
-  ([nuMP| Impl1_Fail str |], _) ->
-    tell [mbLift str] >> mzero
+  ([nuMP| Impl1_Fail str |], _) -> pitmFail $ mbLift str
   
   ([nuMP| Impl1_Catch |],
    [nuMP| (MbPermImpls_Cons _ (MbPermImpls_Cons _ _ mb_impl1) mb_impl2) |]) ->
-    pitmCatching (translatePermImpl prx $ mbCombine RL.typeCtxProxies mb_impl1) >>= \maybe_trans1 ->
-    pitmCatching (translatePermImpl prx $ mbCombine RL.typeCtxProxies mb_impl2) >>= \maybe_trans2 ->
-    case (maybe_trans1, maybe_trans2) of
-      (Just trans1, Just trans2) ->
+    pitmCatching (translatePermImpl prx $
+                  mbCombine RL.typeCtxProxies mb_impl1) >>= \trans_pair1 ->
+    pitmCatching (translatePermImpl prx $
+                  mbCombine RL.typeCtxProxies mb_impl2) >>= \trans_pair2 ->
```

Comment:
Why not use a pattern matching lambda here (i.e. `... >>= \(mtrans1,hasf1) ->`) like you do below (lies 3150 and 3152)?